### ISSUE TITLE
Prevent skipping tests without leaving a special comment

### ISSUE THREAD

**Issue by brian-mann** [cypress-io/eslint-plugin-dev#14]
### Skipped Tests

We should write a custom eslint rule that prevents skipped tests such as the following:

```js
it.skip('does something probably very important, () => {
  // implementation here...
  foo.bar()
  expect(baz).to.eq(quux)
})
```

The rule should require the developer to indicate **why** the test is being skipped such as:

```js
// NOTE: this test is skipped because the implementation has not yet been
// written and is waiting on a refactor from PR #123, and will be finished
// by the time this PR is merged.
it.skip('does something probably very important, () => {
  // implementation here...
  foo.bar()
  expect(baz).to.eq(quux)
})
```

The idea here is that the developer should provide **context** as to why the test is skipped and notes like these will show up in greater apparent detail during reviews of the diffs.

### Comment Implementation

The eslint custom rule should support finding single line comments `//` or multiline comments `/* */` and additionally require a special keyword like `NOTE:`, or some other word to not get confused by a regular comment that just happens to be sitting above the `it` declaration.

It's common to use VS code plugins that automatically highlight certain words like `NOTE` or `TODO`, etc. I know @chrisbreiding 	and I use something similar. Open to suggestions or ideas here.

### Empty Tests

Tests that are skipped due to not having a test body should **not** require a special comment:

```js
it('is not yet implemented')

// no need for a special comment above the test
```

### Bonus Points (optional)

It would be really cool if we could also detect commented out tests - since they are actually the same thing as skipped tests.

For instance this commented out test should also require the developer to provide context:

```js
// it('does something probably very important, () => {
//   // implementation here...
//   foo.bar()
//   expect(baz).to.eq(quux)
// })
```

This should also fail and should look like this instead...

```js
// NOTE: this test is commented out because the implementation 
// has changed but we haven't yet decided on the new specification
// and we're keeping the old tests as reference temporarily
// it('does something probably very important, () => {
//   // implementation here...
//   foo.bar()
//   expect(baz).to.eq(quux)
// })
```

**Comment by bahmutov** [comment]
I have this rule here https://github.com/bahmutov/eslint-rules#no-commented-out-code that detects any commented out JavaScript code, we can do the same - detect "it('..." in the commented out code

**Comment by kuceb** [comment]
done in #15, released in `3.3.0`
